packages/card/src/react/index.tsx

```diff
@@ -0,0 +1,359 @@
+    const label = 'psds-card__action-bar__button'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      disabled && css(stylesheet[`.${label}--disabled`])
+    )
+  },
+
+  bonusBar: () => css(stylesheet['.psds-card__bonus-bar']),
+
+  card: () => css(stylesheet['.psds-card']),
+
+  fullOverlay: ({ fullOverlayVisible: visible }: Partial<CardProps>) => {
+    const label = 'psds-card__full-overlay'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      visible && css(stylesheet[`.${label}--fullOverlayVisible`])
+    )
+  },
+  fullOverlayLink: () => css(stylesheet['.psds-card__full-overlay-link']),
+
+  image: () => css(stylesheet['.psds-card__image']),
+  imageLink: () => css(stylesheet['.psds-card__image-link']),
+
+  metadata: (
+    { size }: Partial<MetaDataProps>,
+    themeName: ValueOf<typeof themeNames>
+  ) => {
+    const label = 'psds-card__metadata'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}--size-${size}`]),
+      css(stylesheet[`.${label}--theme-${themeName}`])
+    )
+  },
+  metadataDatum: () => css(stylesheet['.psds-card__metadata__datum']),
+  metadataDot: () => css(stylesheet['.psds-card__metadata__dot']),
+
+  overlays: ({ size }: Partial<CardProps>) => {
+    const label = 'psds-card__overlays'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}--size-${size}`])
+    )
+  },
+
+  progress: () => css(stylesheet['.psds-card__progress']),
+  progressBar: ({ progress }: Partial<CardProps>) => {
+    const label = 'psds-card__progress__bar'
+    const percent = toPercentageString(progress || 0)
+    const isCompleted = percent === '100%'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      isCompleted && css(stylesheet[`.${label}--complete`]),
+      css({ width: percent })
+    )
+  },
+
+  tag: () => css(stylesheet['.psds-card__tag']),
+  tagIcon: () => css(stylesheet['.psds-card__tag__icon']),
+  tagText: () => css(stylesheet['.psds-card__tag__text']),
+
+  textLink: (themeName: ValueOf<typeof themeNames>) => {
+    const label = 'psds-card__text-link'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}--theme-${themeName}`])
+    )
+  },
+
+  title: (themeName: ValueOf<typeof themeNames>) => {
+    const label = 'psds-card__title'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}--theme-${themeName}`])
+    )
+  },
+
+  titleContainer: ({ size }: Partial<CardProps>) => {
+    const label = 'psds-card__title-container'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}--size-${size}`])
+    )
+  }
+}
+
+interface ActionBarActionProps extends React.HTMLAttributes<HTMLButtonElement> {
+  disabled?: boolean
+  icon?: React.ReactElement // TODO: retype as Icon when it's TS
+  title: string
+}
+interface ActionBarActionStatics {}
```

Comment:
if there are no statics you can type this without the extra overhead of `RefForwardingComponent`. not requesting a change - just letting you know. happy to show the difference

Reply:
Great, yeah, I'd love to see an example.